test/e2e/federated-ingress.go

```diff
@@ -464,17 +515,34 @@ func (j *federationTestJig) waitForFederatedIngress() {
 	}
 	j.address = address
 	framework.Logf("Found address %v for ingress %v", j.address, j.ing.Name)
-	timeoutClient := &http.Client{Timeout: reqTimeout}
-
-	// Check that all rules respond to a simple GET.
-	for _, rules := range j.ing.Spec.Rules {
-		proto := "http"
-		for _, p := range rules.IngressRuleValue.HTTP.Paths {
-			route := fmt.Sprintf("%v://%v%v", proto, address, p.Path)
-			framework.Logf("Testing route %v host %v with simple GET", route, rules.Host)
-			framework.ExpectNoError(pollURL(route, rules.Host, lbPollTimeout, lbPollInterval, timeoutClient, false))
-		}
+
+	client := &http.Client{
+		// This is mostly `http.DefaultTransport` except for the
+		// `TLSClientConfig`. `http.DefaultTransport` cannot be
+		// shallow copied due to the unexported `sync.Mutex` in
+		// the struct.
+		Transport: &http.Transport{
```

Comment:
There should be something better
Does using SetTransportDefaults work? Example usage: https://github.com/kubernetes/kubernetes/blob/8fd414537b5143ab039cb910590237cabf4af783/pkg/probe/http/http.go#L35